### ISSUE TITLE
add possibility to edit a worklog

### ISSUE THREAD

**Issue by sebmeyer** [szymonkozak/tempomat#45]
Currently, whenever there is an issue with a worklog, I need to delete and recreate it. 
It would be great if a worklog could be edited (description, start time, duration, ...) to make it easier to fix typos or mistakes in initial logs.

**Comment by szymonkozak** [comment]
Do you have any propositions about the usage? Interactive command or something else? I don't have any idea for good design for this feature. It would be easy to add something like `tempo edit 123 --description "new description"`, but I don't know if you mean that.

**Comment by sebmeyer** [comment]
Yeah, that's what i envisioned to be honest
`tempo edit <worklog_id>` with options for the attributes that are reasonably likely to be edited:
- issue key or alias
- duration or start/end
- description

In my opinion an interactive command would be messier than this, because it'd have to constantly ask what the user wants to edit.
If I can come up with a better solution I'll comment again.

**Comment by Vivec1990** [comment]
I have tried implementing this.
Is it not possible to move a worklog to another issue via the rest API? If i send an update request as defined in the tempo  api with another issue key than the one already associated with the worklog, it always tells me that the issue does not exist (but it does).

@szymonkozak do you have any experience with this?

**Comment by szymonkozak** [comment]
@Vivec1990 I have the same experience as you. It looks like a bug in the API, you can report it here: https://tempo-io.atlassian.net/servicedesk/customer/portal/6 

They are pretty responsive.

I don't have too much time until February/March, so if you want you can implement this feature and open Pull Request.

**Comment by Vivec1990** [comment]
Okay, so the answer to the inquiry was that this is sadly expected behavior.
As the edit worklog request would take basically the same arguments as adding a new worklog, i'll just go the way that tempo ui is apparently going and delete the old worklog and add a new one.

I'll open a pr for this once it is done.
